pkg/cloud/azure/azure_storage.go

```diff
@@ -54,6 +55,13 @@ const (
 	// AzureEnvironmentKeyParam is the query parameter for the environment name in an azure URI.
 	AzureEnvironmentKeyParam = "AZURE_ENVIRONMENT"
 
+	// AzureClientIDParam is the query parameter for client_id in an azure URI.
```

Comment:
could you add one more sentences on the implication of this param? I.e. if the client specifies this, they have a "secret" credential, as oppose to a "shared" credential? I'm not entirely sure what the implications are here.